plans/LayoutDebugger/Overview.md

```diff
@@ -0,0 +1,43 @@
+
+While the layout inspector project is definitely a step toward the right direction and was proven to be helpful, I think we can do a lot more.  One drawback of that project is that we didn't have a clear integration contract with the core components of the framework.  We ended up introducing lots of complexities to ASDisplayNode that were hard to reason and mantain.  Another problem is it was difficult to bootstrap the tool into existing code/layout.
+
+This project aims to provide a functional layout debugging solution that requires minimal change to the core framework, is easy to setup and works out-of-the-box on existing applications.  These will be achieved by building an extension framework that is hosted in a separate repository, integrates with Texture core and leverages Chrome DevTools.
+
+## Execution plan
+
+At first, we'll build on top of [PonyDebugger](https://github.com/square/PonyDebugger).  I played with it for a few hours and [the result](https://www.dropbox.com/s/8bcpdgogoewmox9/view%20debugger.mov?dl=0) looks promising.  The bottom line is that we'll implement a new `PDDomainController` that is inspired by `PDDOMDomainController` but is catered toward `ASLayoutElement`.  The controller should expose style properties, allow editting those properties and, as a result, support hot reloading.  It should also expose the constrained size passed to each element during the previous layout pass.
+
+Eventually, we'll possibly move away from PonyDebugger and implement our own framework due to a few reasons:
+
+1. PonyDebugger is not actively maintained.  That might be because it's considered a "done" project, although the amount of open issues may suggest otherwise.
+2. It's not easy to setup the environment, especially because of the [ponyd gateway server](https://github.com/square/PonyDebugger/tree/master/ponyd).  Ponyd is essentially a middleman that sits between the client code and Chrome DevTools.  It is implemented in Python and hosts its own version of DevTools.  Its bootstrap script is more or less broken.  It's not trivial (at least for me) to setup a working environment.  Instead, my limitted research showed that we can do better by letting the client app be a mDNS broadcaster and allowing Chrome DevTools to connect to it.  The workflow will be very straight-forward for developers, similar to [Stetho](https://facebook.github.io/stetho/)'s.  In addition, the whole project will be simpler because we don't need to maintain ponyd.
```

Comment:
Yes, I'll put up a document once I have a clear idea of what I did to make it work. I tried different suggestions from Google, some might be unnecessary, until I hit no error LOL